db/merge_helper.cc

```diff
@@ -138,7 +138,11 @@ Status MergeHelper::MergeUntil(InternalIterator* iter,
   // orig_ikey is backed by original_key if keys_.empty()
   // orig_ikey is backed by keys_.back() if !keys_.empty()
   ParsedInternalKey orig_ikey;
-  ParseInternalKey(original_key, &orig_ikey);
+  bool succ = ParseInternalKey(original_key, &orig_ikey);
+  assert(succ);
```

Comment:
If `!succ` and `ifndef NDEBUG`, the process will die here. Is this intended?

Reply:
we're probably doomed if this happen anyways. the scenarios I can think of are:

- memory corruption
- the SST was written with an invalid key

A normal corruption should encounter a checksum verification error, not this problem. So I'm ok with handling it aggressively.

Reply:
I'm OK with this too. Just curious whether some static check will fail because it identifies unreachable code.

Reply:
Yes. If this returns false during test, it is likely to be a software issue rather than a hardware failure. Then it is better to catch it as early as possible.